src/Data/WideWord/Word128.hs

```diff
@@ -434,6 +456,67 @@ pokeElemOff128 ptr idx (Word128 a1 a0) = do
   pokeElemOff (castPtr ptr) (2 * idx + index0) a0
   pokeElemOff (castPtr ptr) (2 * idx + index1) a1
 
+-- -----------------------------------------------------------------------------
+-- Functions for `Prim` instance.
+
+{-# INLINE sizeOf128# #-}
+sizeOf128# :: Word128 -> Int#
+sizeOf128# _ = 2# *# sizeOf# (undefined :: Word64)
+
+{-# INLINE alignment128# #-}
+alignment128# :: Word128 -> Int#
+alignment128# _ = alignment# (undefined :: Word64)
+
+{-# INLINE indexByteArray128# #-}
+indexByteArray128# :: ByteArray# -> Int# -> Word128
+indexByteArray128# arr# i# =
+  let x = indexByteArray# arr# (2# *# i#)
+      y = indexByteArray# arr# (2# *# i# +# 1#)
+  in Word128 x y
```

Comment:
Reading a array read from disk, that will give different results on a big endian vs a little endian machine.

Have a look at the way I did `peek128` and `poke128`.